Applications/MIDIMonitor/SMMAppController.m

```diff
@@ -332,9 +331,21 @@ - (void)failedToConnectToSpyClient
 
 - (void)sourceEndpointsAppeared:(NSNotification *)notification
 {
-    if ([[NSUserDefaults standardUserDefaults] boolForKey:SMMOpenWindowsForNewSourcesPreferenceKey]) {
+    NSInteger option = [[NSUserDefaults standardUserDefaults] integerForKey:SMMAutoConnectNewSourcesPreferenceKey];
+    
+    if (option == 1) {
         NSArray *endpoints = [[notification userInfo] objectForKey:SMMIDIObjectsThatAppeared];
-
+        
+        if (!self.newlyAppearedSources) {
+            self.newlyAppearedSources = [NSMutableSet set];
+            [self performSelector:@selector(autoConnectToNewlyAppearedSources) withObject:nil afterDelay:0.1 inModes:@[NSDefaultRunLoopMode]];
```

Comment:
Since the code for cases 1 and 2 are exactly the same, aside from the `@selector`, it would be better to avoid repeating the code. Move the check for option == 1 or option == 2 down here, and use it to choose a different selector.